libdrgn/object.c

```diff
@@ -1135,31 +1135,31 @@ drgn_object_reinterpret(struct drgn_object *res,
 LIBDRGN_PUBLIC struct drgn_error *
 drgn_object_integer_literal(struct drgn_object *res, uint64_t uvalue)
 {
-	const struct drgn_language *lang = &drgn_language_c;
+	const struct drgn_language *lang = drgn_object_lang(res);
```

Comment:
`drgn_object_*_literal()` don't operate on the existing object but rather replace it, so this should be `drgn_program_lang(obj->prog)` here and in the other literal helpers.